Project.toml

```diff
@@ -1,14 +1,15 @@
 name = "SimpleNonlinearSolve"
 uuid = "727e6d20-b764-4bd8-a329-72de5adea6c7"
 authors = ["SciML"]
-version = "0.1.11"
+version = "0.1.12"
 
 [deps]
 ArrayInterfaceCore = "30b0a656-2188-435a-8636-2ec0e6a096e2"
 DiffEqBase = "2b5f629d-d688-5b77-993f-72d75c75574e"
 FiniteDiff = "6a86dc24-6348-571c-b903-95158fe2bd41"
 ForwardDiff = "f6369f11-7733-5829-9624-2563aa707210"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+NNlib = "872c559c-99b0-510c-b3b7-b6c96a88d5cd"
```

Comment:
oh definitely not in this library 😅 Maybe that needs to be an extension.

Reply:
:sweat:  I guessed so. Let me set it up as a Pkg Extension then

Reply:
For Pkg Extension, I think we will have to set it up as a dispatch on algorithm and not a keyword argument right? Something like `Broyden{false}` by default and `Broyden{true}` for the batched version

Reply:
I think so.

Reply:
yes it probably needs to be split like that. What is the NNlib part for? Why is the matmul batched instead of just a single matmul?

Reply:
The matmul is `A x B x Batch` * `B x C x Batch`, so they are independent matmuls along the batch dimension. Also IIRC the NNlib implementation does proper CUBLAS dispatches https://github.com/FluxML/NNlibCUDA.jl/blob/5f797aec23cbb5483788697e9e911685b681bbf8/src/batchedmul.jl#L3